pkg/kubectl/cmd/util/openapi/openapi.go

```diff
@@ -130,21 +132,45 @@ type Type struct {
 	// Extensions are extensions for this field and may contain
 	// metadata from the types.go struct field tags.
 	// e.g. contains patchStrategy, patchMergeKey, etc
-	Extensions spec.Extensions
+	Extensions map[string]interface{}
+}
+
+func vendorExtensionToMap(e []*openapi_v2.NamedAny) map[string]interface{} {
+	var values map[string]interface{}
+
+	for _, na := range e {
+		if na.GetName() == "" || na.GetValue() == nil {
+			continue
+		}
+		if na.GetValue().GetYaml() == "" {
+			continue
+		}
+		var value interface{}
+		err := yaml.Unmarshal([]byte(na.GetValue().GetYaml()), &value)
+		if err != nil {
+			continue
+		}
+		if values == nil {
+			values = make(map[string]interface{})
+		}
+		values[na.GetName()] = value
+	}
+
+	return values
 }
 
 // NewOpenAPIData parses the resource definitions in openapi data by groupversionkind and name
-func NewOpenAPIData(s *spec.Swagger) (*Resources, error) {
+func NewOpenAPIData(doc *openapi_v2.Document) (*Resources, error) {
 	o := &Resources{
 		GroupVersionKindToName: map[schema.GroupVersionKind]string{},
 		NameToDefinition:       map[string]Kind{},
 	}
 	// Parse and index definitions by name
-	for name, d := range s.Definitions {
-		definition := o.parseDefinition(name, d)
-		o.NameToDefinition[name] = definition
+	for _, ns := range doc.GetDefinitions().GetAdditionalProperties() {
+		definition := o.parseDefinition(ns.Name, ns.Value)
```

Comment:
does ns have Getters for name and value. If yes, then we should use those to be consistent.